src/mpc-hc/SubtitleDlDlg.cpp

```diff
@@ -20,301 +20,131 @@
  */
 
 #include "stdafx.h"
-#include <afxwin.h>
 #include "SubtitleDlDlg.h"
-#include "MainFrm.h"
-#include "DSUtil.h"
+#include "SubtitlesProvider.h"
 
 // User Defined Window Messages
-#define UWM_PARSE  (WM_USER + 100)
-#define UWM_FAILED (WM_USER + 101)
-
-CSubtitleDlDlg::CSubtitleDlDlg(CWnd* pParent, const CStringA& url, const CString& filename)
-    : CResizableDialog(CSubtitleDlDlg::IDD, pParent)
-    , m_ps(nullptr, 0, TRUE)
-    , m_defps(nullptr, filename)
-    , m_pTA(nullptr)
-    , m_url(url)
+enum {
+    UWM_SEARCH = WM_USER + 100,
+    UWM_SEARCHING,
+    UWM_DOWNLOADING,
+    UWM_DOWNLOADED,
+    UWM_COMPLETED,
+    UWM_FINISHED,
+    UWM_FAILED,
+    UWM_CLEAR,
+};
+
+CSubtitleDlDlg::CSubtitleDlDlg(CWnd* pParentWnd)
+    : CResizableDialog(CSubtitleDlDlg::IDD, pParentWnd)
+    , m_ps(nullptr, 0, 0)
     , m_fReplaceSubs(false)
-    , m_status()
+    , m_pMainFrame(*(CMainFrame*)(pParentWnd))

```

Comment:
If `m_pMainFrame` is reference it shouldn't be prefixed with `p` which is for pointers.